libsupport/include/katana/Result.h

```diff
@@ -230,22 +282,54 @@ class KATANA_EXPORT CopyableErrorInfo {
   std::ostream& Write(std::ostream& out) const;
 
 private:
+  template <typename F, typename... Args>
+  void PrependFmt(F fmt_string, Args&&... args) {
+    std::vector<char> out;
+    fmt::format_to(
+        std::back_inserter(out), fmt_string, std::forward<Args>(args)...);
+    Prepend(out.data(), out.data() + out.size());
+  }
+
+  void Prepend(const char* begin, const char* end) {
+    if (!message_.empty()) {
+      message_.insert(0, std::string(": "));
+    }
+    message_.insert(message_.begin(), begin, end);
+  }
+
   std::error_code error_code_;
   std::string message_;
 };
 
+#define KATANA_COPYABLE_ERROR(ec, fmt_string, ...)                             \
+  ::katana::CopyableErrorInfo::MakeWithSourceInfo(                             \
+      __FILE__, __LINE__, (ec), FMT_STRING(fmt_string), ##__VA_ARGS__)
```

Comment:
This currently returns `ErrorInfo`, but works anyway because of implicit conversions. See main comment.